### PR TITLE
Improve docs of dropout

### PR DESCRIPTION
**Pull request by naoto0804**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:
 - functions/noise/dropout

### CODE REVIEW

**Comment by naoto0804** [comment]
screenshot

**Comment by naoto0804** [comment]
[screenshot.pdf](https://github.com/chainer/chainer/files/1204397/screenshot.pdf)

**Comment by naoto0804** [comment]
screenshot

**Comment by naoto0804** [comment]
screenshot

File: chainer/functions/noise/dropout.py [hunk 1]
```diff
@@ -63,7 +63,9 @@ def dropout(x, ratio=.5, **kwargs):
        See :func:`chainer.using_config`.
 
     Args:
-        x (~chainer.Variable): Input variable.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Input variable. \
+        A :math:`(s_1, s_2, ..., s_N)` -shaped float array.
```
Comment (keisuke-umezawa) [inline chainer/functions/noise/dropout.py]: The comment is ok.  But it seems to make some indent error. You do not need to put the backslash after the "Input variable" and have to put the indent before "A math". Please see detail if https://github.com/chainer/chainer/blob/master/chainer/functions/activation/sigmoid.py

**Comment by naoto0804** [comment]
screenshot_2

File: chainer/functions/noise/dropout.py (hunk 1 above)
Comment (naoto0804) [inline chainer/functions/noise/dropout.py]: Fixed.

Comment (naoto0804) [inline chainer/functions/noise/dropout.py]: Fixed

**Comment by naoto0804** [comment]
I've also fixed indentation error probably caused by #3100 (please see a description of `ratio` in  two screenshots).

**Comment by keisuke-umezawa** [comment]
jenkins, test this please

**Comment by keisuke-umezawa** [comment]
@mitmul How about this PR? I reviewed this PR and approved. Please merge this PR.

**Comment by ronekko** [comment]
Hi.
In testing mode, the type of the return value is `ndarray` when the input is `ndarray`.

```python
import numpy as np
import chainer

x = numpy.ones(5, 'f')
with chainer.using_config('train', False):
    y = chainer.functions.dropout(x)
    print(type(y))  # <class 'numpy.ndarray'>
    y = chainer.functions.dropout(chainer.Variable(x))
    print(type(y))  # <class 'chainer.variable.Variable'>
```

**Comment by naoto0804** [comment]
@ronekko Thank you! I'll send another PR to fix it before this PR.

**Comment by naoto0804** [comment]
screenshot3

**Comment by naoto0804** [comment]
Bug fix in `functions.noise.dropout` was merged in PR #3239.
I reflect this change and fix admonition.
Could you check it again?

**Comment by keisuke-umezawa** [comment]
jenkins, test this please

**Comment by keisuke-umezawa** [comment]
LGTM！ Wait for the check of Jenkins test.

**Comment by keisuke-umezawa** [comment]
Jenkins, test this please

**Comment by keisuke-umezawa** [comment]
@mitmul I can not kick the test. Please start the test and merge this PR.

**Comment by mitmul** [comment]
Jenkins, test this please

**Comment by mitmul** [comment]
Jenkins...Jenkins... test this please

**Comment by mitmul** [comment]
LGTM

**Pull request merged**